packages/api/src/Base.ts

```diff
@@ -532,6 +532,25 @@ export default abstract class ApiBase<CodecResult, SubscriptionResult> implement
     decorated.key = (arg?: CodecArg): string =>
       u8aToHex(compactStripLength(method(arg))[1]);
 
+    // Linked Map support
+
+    if (method.headKey) {
+      decorated.head = (): C =>
+        onCall(
+          (arg: CodecArg) => this._rpcRx.state
+            .getStorage(arg)
+            .pipe(
+              switchMap(key => this._rpcRx.state.getStorage([method, key]))
```

Comment:
this needs to return both the value and the key. but I don't think I can return two values with Observable? Should I return `Tuple` instead?

Reply:
Yes, Tuple makes sense so it is still a `Codec`